psd-web/app/models/user.rb

```diff
@@ -17,21 +17,21 @@ def self.activated
     where(account_activated: true)
   end
 
-  def self.create_and_send_invite(email_address, team, redirect_url)
-    KeycloakClient.instance.create_user(email_address)
-
-    keycloak_user = KeycloakClient.instance.get_user(email_address)
-
-    # Create the user in the local cache database so that we don't have to wait until the next sync
-    user = create(id: keycloak_user[:id], email: email_address, organisation: team.organisation)
+  def self.create_and_send_invite(email_address, team, inviting_user)
+    user = create(
+      id: SecureRandom.uuid,
+      email: email_address,
+      organisation: team.organisation,
+      invitation_token: SecureRandom.hex(15)
+    )
     team.add_user(user)
 
-    KeycloakClient.instance.send_required_actions_welcome_email keycloak_user[:id], redirect_url
+    SendUserInvitationJob.perform_later(user.id, inviting_user.id)
   end
 
-  def self.resend_invite(email_address, _team, redirect_url)
-    user_id = KeycloakClient.instance.get_user(email_address)[:id]
-    KeycloakClient.instance.send_required_actions_welcome_email user_id, redirect_url
+  def self.resend_invite(email_address, inviting_user)
+    user = User.find_by!(email: email_address)
```

Comment:
Not a biggie but we might want to check the user belongs to the same team as `inviting_user` here by adding an extra condition.

Reply:
```suggestion
    user = inviting_user.teams.find_by!(email: email_address)
```

Reply:
Good point! The existing code doesn't do this though, so can we raise this as a separate ticket?

Reply:
It’s probably a very simple fix so we should be able to do it now.

Reply:
Well one complication is that users can currently belong to multiple teams – although in practice this hasn't happened, and we might remove this ability in future.

So the simplest fix would be something like

```ruby
user = inviting_user.teams.first.users.find_by!(email: email_address)
```

but in theory in should be something more like

```ruby
user = User.find_by!(email: email_address)
raise ActiveRecord::RecordNotFound unless user.teams & inviting_user.teams
```
(ie checking the intersection of the two users’ teams)

Reply:
this would always raise

Reply:
```suggestion
    user = User.where(teams: inviting_user.teams).find_by!(email: email_address)
```
this is probably what you'd want.